packages/solo/src/web.js

```diff
@@ -148,21 +149,28 @@ export async function makeHTTPListener(
   const htmldir = path.join(basedir, 'html');
   log(`Serving static files from ${htmldir}`);
   app.use(express.static(htmldir));
-  app.use(express.static(new URL('../public', import.meta.url).pathname));
+  const soloHtmlDir = new URL('../public', import.meta.url).pathname;
+  app.use(express.static(soloHtmlDir));
 
   if (walletHtmlDir && !bridgeTarget) {
     // Transition localStorage based bridge
+    console.log('redirecting wallet bridge');
     if (existsSync(path.join(walletHtmlDir, 'bridge.html'))) {
-      console.log('redirecting wallet bridge');
       app.get('/wallet-bridge.html', (req, res) =>
-        res.redirect('/wallet/bridge.html'),
+        res.redirect(
+          url.format({ pathname: '/wallet/bridge.html', query: req.query }),
+        ),
+      );
+    } else {
+      app.get('/wallet/bridge.html', (_, res) =>
+        res.sendFile(path.resolve(soloHtmlDir, 'wallet-bridge.html')),
```

Comment:
Why not redirect to `wallet-bridge.html` ?

Reply:
No redirect because I want `/wallet/bridge.html` to be canonical.